README.md

```diff
@@ -28,14 +28,222 @@ In addition, that format would include optional **signing** of the resources, wh
+   number to the last of the final magic number) as a little-endian 8-byte
+   integer.
+3. A 1-byte count of metadata sections followed by that many of the following
+   entries. Repeated entry IDs and non-monotonically-increasing offsets cause
+   the package to fail to parse.
+   * The byte `01` indicating an [index](#index), followed by the offset of the
+     start of the index.
+   * The byte `02` indicating a block of [certificates](#certificates), followed
+     by the offset of the start of the certificates.
+   * The byte `03` indicating a [manifest](#manifest), followed by the offset of
+     the start of the manifest.
+
+   Unknown entry IDs are assumed to be followed by an offset, and the pair must
+   be ignored.
+4. The offset of the [main content](#main-content) of the package.
+5. The metadata sections promised above.
+6. A sequence of HTTP responses constituting the main content of the package.
+   Mismatches with the manifest usually cause the package to fail to parse, as
+   described in the [manifest](#manifest) section, but mismatches with the index
+   might not.
+7. The length of the package, repeated from earlier. This allows tools to
+   operate on the package even if it's been prepended with a self-extracting
+   executable.
+8. The same magic number: `F0 9F 8C 90 F0 9F 93 A6` (🌐📦 in UTF-8).
+
+The MIME type of this format is `application/package`. TODO: Is that right?
+
+Note that this top-level information is *not signed*, and so can't be trusted.
+Only information in the manifest and below can be trusted.
+
+The manifest and certificates are used to sign the contents of the package. The
+index is used when the package is stored on disk, to find resources quickly. And
+the start page should be loaded when a UA is navigated to the package itself.
+
+### Index
+
+The index is optional, and can be added by a client after the package is
+downloaded.
+
+It contains a series of lines (terminated by `\r\n`) of the form
 
-We also propose to remove the [fragment-based URL schema](https://w3ctag.github.io/packaging-on-the-web/#fragment-identifiers) from the spec as it is not clear what would be the use case supporting it.
+```
+URL offset length
+```
+
+Where offset and length are 64-bit integers, and the URLs are absolute. Offsets
+are interpreted relative to the first byte of the package. All resources in the
+package should be included in the index, even if they come from a different
+origin.
+
+TODO: Provide base URLs, maybe as INI-format sections.
+
+```
+https://example.org/index.html 7000 500
+https://example.org/img.png 7500 1000
+...
+```
+
+### Certificates
+
+The certificates block contains
+an
+[application/pkcs7-mime; smime-type=certs-only](https://tools.ietf.org/html/rfc5751) resource
+holding enough [X.509](https://tools.ietf.org/html/rfc5280) certificates that
+chains can be built from the certificates that sign the top-level
+and [sub-package](#sub-packages) manifests to root certificates recognized by
```

Comment:
This makes it not clear what certificates and for what signature are included. The original idea went with a packages being authonomous units (since if signed, they most likely will be signed by separate domain owners, using separate certs etc). It's unclear why including certs needed to verify a subpackage is needed in a containing package, if that is what you meant.